src/Providers/CpServiceProvider.php

```diff
@@ -44,6 +45,14 @@ public function boot()
         Sets::setIconsDirectory();
 
         $this->registerMiddlewareGroups();
+
+        if (method_exists(AuthenticateSession::class, 'redirectUsing')) {
+            AuthenticateSession::redirectUsing(function () {
+                return config('statamic.cp.auth.enabled', true)
+                    ? cp_route('login')
+                    : config('statamic.cp.auth.redirect_to', '/');
+            });
+        }
```

Comment:
This is still going to redirect to the CP on a non-CP page.

I meant if you use the middleware on a front-end route you don't want to get taken to the CP. In those cases, we want to just _not_ call this method, and let them deal with it themselves. e.g. Keep the `login` route or have them adjust it.

Maybe we can have another earlier middleware that gets run for CP routes that calls this method.